controllers/users.js

```diff
@@ -0,0 +1,166 @@
+      return res.status(409).json({ message: "User already exists" });
+    }
+
+    try {
+      const newUser = new User({ email, avatarURL });
+      await newUser.setPassword(password);
+      await newUser.save();
+      res.status(201).json({
+        message: "User created",
+        data: {
+          user: {
+            email,
+            subscription: "starter",
+          },
+        },
+      });
+    } catch (e) {
+      next(e);
+    }
+  } else {
+    return res.status(400).json({ message: error.message });
+  }
+};
+
+// =================== LOGIN USER ===================
+const loginUser = async (req, res, next) => {
+  const { email, password } = req.body;
+  const { error } = userSchema.validate({ email, password });
+
+  if (!error) {
+    const user = await User.findOne({ email });
+    const isPasswordCorrect = await user.validatePassword(password);
+
+    if (!user || !isPasswordCorrect) {
+      return res.status(401).json({ message: "wrong credentials" });
+    }
+
+    const payload = {
+      id: user._id,
+      email: user.email,
+    };
+
+    const token = jwt.sign(payload, process.env.SECRET, { expiresIn: "1h" });
+
+    res.status(200).json({
+      token,
+      user: {
+        email,
+        subscription: user.subscription,
+        id: user._id,
+      },
+    });
+  } else {
+    return res.status(400).json({ message: error.message });
+  }
+};
+
+// =================== CURRENT USER ===================
+const currentUser = async (req, res, next) => {
+  const { _id } = req.user;
+  try {
+    const user = await User.findOne({ _id });
+    res.status(200).json({
+      message: "You are authorized",
+      data: {
+        email: user.email,
+        subscription: user.subscription,
+      },
+    });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+// =================== LOGOUT USER ===================
+const logoutUser = async (req, res, next) => {
+  const { _id } = req.user;
+  try {
+    await service.updateUserJWT(_id, null);
+    return res.status(204).json({
+      status: "No Content",
+    });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+// =================== UPDATE AVATAR ===================
+const updateAvatar = async (req, res, next) => {
+  const { _id } = req.user;
+  const avatarURL = `./avatars/av_${_id}.png`;
+  jimp
+    .read(`tmp/${req.file.filename}`)
+    .then((avatar) => {
+      return avatar.resize(250, 250).write(`public/avatars/av_${_id}.png`);
+    })
+    .catch((e) => {
+      console.error(e);
+    });
+  try {
+    const result = await service.updateUserAvatar(_id, avatarURL);
+    if (result) {
+      fs.unlink(req.file.path, (e) => {
+        console.error(e);
+      });
+      res.status(200).json({ avatarURL });
+    } else {
+      res.status(404).json({ message: "User not found" });
+    }
+  } catch (error) {
+    console.error(e);
+    next(e);
+  }
+};
+
+// =================== LIST ALL USERS ===================
```

Comment:
Myślę że jak mamy metode listAllUser to chyba zbędne jest dodawania komentarza?

Reply:
poprawione :) (wprowadziłem żeby łatwiej mi było znaleźć początek każdej funkcji)